custom-transform/app.R

```diff
@@ -0,0 +1,137 @@
+        iv$enable()
+        
+        reactive_datanames <- reactive({
+          req(data())
+          teal.data::datanames(data())
+        })
+        observeEvent(reactive_datanames(), {
+          selected_a <- isolate(input$merge_a)
+          if (identical(selected_a, "")) selected_a <- restoreInput(session$ns("merge_a"), NULL)
+          teal.widgets::updateOptionalSelectInput(
+            session = session,
+            inputId = "merge_a",
+            choices = reactive_datanames(),
+            selected = restoreInput(session$ns("merge_a"), selected_a)
+          )
+
+          selected_b <- isolate(input$merge_b)
+          if (identical(selected_b, "")) selected <- restoreInput(session$ns("merge_b"), NULL)
+          teal.widgets::updateOptionalSelectInput(
+            session = session,
+            inputId = "merge_b",
+            choices = reactive_datanames(),
+            selected = restoreInput(session$ns("merge_b"), selected_b)
+          )
+        })
+        
+        merge_a <- reactive(input$merge_a)
+        merge_b <- reactive(input$merge_b)
+        
+        reactive({
+          new_data <- within(
+            data(),
+            ANL <- dplyr::left_join(merge_a, merge_b),
+            merge_a = tryCatch(as.name(merge_a()), error = function(e) as.name("DatasetA")),
+            merge_b = tryCatch(as.name(merge_b()), error = function(e) as.name("DatasetA"))
+          )
+          teal.data::datanames(new_data) <- c(teal.data::datanames(new_data), "ANL")
+          new_data
+        })
+      })
+    }
+  )
+)
+
+data <- teal_data() %>%
+  within(
+    {
+      ADSL <- teal.data::rADSL
+      ADTTE <- teal.data::rADTTE
+      iris <- iris
+
+      CO2 <- CO2
+      factors <- names(Filter(isTRUE, vapply(CO2, is.factor, logical(1L))))
+      CO2[factors] <- lapply(CO2[factors], as.character)
+    }
+  )
+join_keys(data) <- default_cdisc_join_keys[c("ADSL", "ADTTE")]
+teal.data::datanames(data) <- c("ADSL", "ADTTE", "iris", "CO2")
+
+nest_logo <- "https://raw.githubusercontent.com/insightsengineering/hex-stickers/main/PNG/nest.png"
+app_source <- "https://github.com/insightsengineering/teal.gallery/tree/main/basic-teal"
+gh_issues_page <- "https://github.com/insightsengineering/teal.gallery/issues"
+
+header <- tags$span(
+  style = "display: flex; align-items: center; justify-content: space-between; margin: 10px 0 10px 0;",
+  tags$span("Teal app with custom transform", style = "font-size: 30px;"),
+  tags$span(
+    style = "display: flex; align-items: center;",
+    tags$img(src = nest_logo, alt = "NEST logo", height = "45px", style = "margin-right:10px;"),
+    tags$span(style = "font-size: 24px;", "NEST @ Roche")
+  )
+)
+
+footer <- tags$p(
+  "This teal app is brought to you by the NEST Team at Roche/Genentech.
+        For more information, please visit:",
+  tags$a(href = app_source, target = "_blank", "Source Code"), ", ",
+  tags$a(href = gh_issues_page, target = "_blank", "Report Issues")
+)
```

Comment:
This is not a comment for this PR, but all apps code is extended by 20 lines just to add a NEST logo
general issue here: https://github.com/insightsengineering/teal.gallery/issues/178